### DISCUSSION TITLE
Miscellaneous questions and ideas

### DISCUSSION THREAD

**Discussion by ChanceHarrison** [woelper/oculante#232] [General]
Hi there! I just discovered Oculante and I think it's really awesome. I had some initial questions and ideas that I wanted to share:

1. I was hoping to use Oculante to read image data from my clipboard, but when I opened Oculante and pressed `Ctrl+V`, nothing happened? Is this expected? The keybindings section has an option for "Paste" so I thought it would be. Is it only supported in edit mode? When I went to the menu and clicked "Paste" error text appeared at the bottom of the window saying "Clipboard did not contain image" though it distinctly does. Maybe it's because I have a non-standard desktop environment/clipboard setup? What is necessary to support clipboard operations with Oculante?
2. I noticed that Oculante has a network receive mode, and that's really cool! I was hoping I could use Oculante in a similar manner to read image data that is piped into it, though that doesn't seem to be supported?
3. When I clicked "Browse for Image," Oculante exited with the following output:
    ```
    (oculante:119760): Gtk-WARNING **: 03:41:26.220: Unknown key Settings in /home/ch/.config/gtk-3.0/settings.ini
    (oculante:119760): GLib-GIO-ERROR **: 03:41:26.280: No GSettings schemas are installed on the system
    ``` 
    I suppose the error is clear enough, and I can only imagine that Oculante makes some basic assumtpions about a user's operating environment that I am not adhering to, but for the program to forcefully exit when I try to open a file picker in this situation is a bit...harsh. Is it possible to change that behavior? And in the meantime, I suppose I need to figure out what I need to add to my minimal desktop environment to make file pickers work (probably some kind of file browser program and some kind of GTK/GLib configuration based on the error message). Maybe once I figure out there could be an opportuntiy to contribute back what I learned in documentation somehow?

4. When I launched `oculante` from a shell, I was expecting it to launch and go into the background or otherwise fork away instead of holding my terminal foreground. I am aware I can use my shell's tools (like described [here](https://unix.stackexchange.com/questions/145367/how-to-have-back-the-command-prompt-after-calling-an-application) and [here](https://unix.stackexchange.com/questions/103731/run-a-command-without-making-me-wait?noredirect=1&lq=1)) to background the process or otherwise solve this issue myself, but I was thinking that unless Oculante needs to hold the foreground for some reason, would it be possible to change the default behavior in this regard?
5. Is there something I'm missing regarding the "Compare" functionality in the "Info" panel? I figured out that I could add the open image to the list, switch to a different image, and add that one to the list as well, and that I could click on an image in the list to switch to it or use the appropriate keybinding, but I was left wondering "is this it?". It seemed no different than if I just used the left and right arrow keys to switch between two or more images. I guess that method depends on the images being next to each other in the opened directory, so I can see how the compare functinoality is an upgrade in that regard. But, other than that, is that all it does?
6. I tried to scroll horizontally using `Shift+Scroll` but it just treats it like a vertical scroll. I figured out that I can use `Shift+<Arrow>` to move the image in the corresponding direction (though it feels inverted! I wish there was a way to change that). Is horizontal scroll with the `Shift` modifier something of interest?
7. I tried enabling "Do not reset image view" in the preferences, but when I closed the preferences window to test it, it didn't work. And when I open the preferences again, indeed it shows as disabled. This seems like uninteiontal behavior. Should I open an issue for this? It doesn't seem like you have a bug report template or guidelines on what information I should include, so if you were to share that I can get that issue started off on the right foot!
8. Do you have an interest in outside contributions? I didn't find a `CONTRIBUTING.md` in the wiki, so I'm curious if you're open to contributions in the first place and if you are what kind of process should be followed to do so.

I know that is a lot, but I look forward to any thoughts you have to share and I hope you might find some of this useful! Thanks for your time!

**Comment by woelper** [comment]
Hi, and many many thanks for your nice words and questions!
1. Oculante should absolutely support pasting from the clipboard. Could you open an issue for that please? I can also look at providing better error messages. If I recall correctly, I am using the `image` library to decode the data into an image. That can of course fail if the data is somehow malformed. You could try pasting your data into another app to check - also you can copy an image in oculante, switch image and paste again to see if there is something completely off.
2. You mean to stdin or such? It would be quite easy to add that. What is your desired use case and how would you like it to behave?
3. Oops! That should never happen. I suspect that is happening within `rfd`, the crate providing the file picker. I have started work on my own file browser that would fix that, but it is not yet done. Of course contributions are always welcome.
4. It would be possible I guess. This may conflict with your point `2.` above, but just maybe. I don't think many people use it from a terminal, and I am absolutely open to have it fork on launch.
5. Pretty much :) This was a user request, and if you are missing something, let me know please! The only upgrade is that you can align images perfectly, store them as a comparison and flick quickly between them.
6. Yes! You are just the fist one asking I guess. Should not be hard to do.
7. Please open an issue if you think that is unintended, and it seems like that. Don't worry about guidelines, just add some info how this can be reproduced as best as possible.
8.  There have been some contributors in the past, and I do have interest! Thank you for asking. There are no guidelines, although maybe there should. First and foremost this should be a tool that is functional, fast and fun to use and is a hobby project of mine. As long as you are OK with casual, relaxed and friendly communication you are welcome!

Thank you so much - questions and feedback like this make this a better tool for everyone!

**Comment by woelper** [comment]
On 6.) - you can configure the keybindings in the settings menu in case something feels inverted.
On 7.) - I was not able to reproduce that. When I enable "Do not reset image view" and immediately selecting the next image, the view is not reset. This should also persist between restarts. I think in your case even the checkbox resets?
